### PR TITLE
Linuxbrew fixes for terraform and packer.

### PR DESCRIPTION
**Pull request by hartzell**
Add logic for checking platform and setting arch and os values appropriately.

My test box (CentOS 7 VM on Digital Ocean, minimal rpms installed) requires that mercurial depend on python.

Tested on CentOS7 on Digital Ocean and on a Mac running 10.11.2.

### CODE REVIEW

**Comment by DomT4** [comment]
Linux-exclusive formulae fixes should be submitted to [Linuxbrew](https://github.com/Linuxbrew/linuxbrew), Thanks!

**Comment by hartzell** [comment]
Apologies.  I was following the recommendations in [Linuxbrew's Contributing.md document](https://github.com/Linuxbrew/linuxbrew/blob/master/CONTRIBUTING.md).

These aren't linux exclusive fixes, they're changes to the existing Homebrew formulas so that they build on either platform.

**Comment by sjackman** [comment]
Hi, George. I'll reword the CONTRIBUTING document. Patches to fix build issues that only affect Linux should be sent to Linuxbrew. Patches to fix build issues that affect both Mac and Linux should be sent to Homebrew.

**Comment by sjackman** [comment]
Here's the improved text. I hope that it's more clear.

> Homebrew is merged into Linuxbrew roughly once per week. If you wish to contribute a new formula, a new version of an existing formula, or any other change that is not specific to Linux, please send your pull request to Homebrew rather than to Linuxbrew. Patches to fix build issues that affect both Mac and Linux should be sent to Homebrew. Patches to fix build issues that affect only Linux should be sent to Linuxbrew.

**Comment by mikemcquaid** [comment]
> Patches to fix build issues that affect both Mac and Linux should be sent to Homebrew.

I might reword this to:
"Patches to fix issues that you have reproduced on both Linuxbrew and Homebrew should be sent to Homebrew."

**Comment by sjackman** [comment]
Done. The revised text is

> Linuxbrew is a fork of Homebrew. Homebrew is merged into Linuxbrew roughly once per week. To contribute a new formula or a new version of an existing formula, please submit your pull request to Homebrew rather than to Linuxbrew. Patches to fix issues that you have reproduced on both Linuxbrew and Homebrew should be sent to Homebrew. Please send your pull request to Linuxbrew if you are in doubt.

**Comment by mikemcquaid** [comment]
@sjackman Perfect, thanks.
